packages/ng-query/src/lib/base-query.ts

```diff
@@ -0,0 +1,82 @@
+import {
+  DefaultedQueryObserverOptions,
+  notifyManager,
+  QueryKey,
+  QueryObserver,
+  QueryObserverResult,
+} from '@tanstack/query-core';
+import { Observable } from 'rxjs';
+
+export function notify<
+  TQueryFnData = unknown,
+  TError = unknown,
+  TData = TQueryFnData,
+  TQueryData = TQueryFnData,
+  TQueryKey extends QueryKey = QueryKey
+>(
+  options: DefaultedQueryObserverOptions<
+    TQueryFnData,
+    TError,
+    TData,
+    TQueryData,
+    TQueryKey
+  >
+) {
+  options.onError &&= notifyManager.batchCalls(options.onError);
+
+  options.onSuccess &&= notifyManager.batchCalls(options.onSuccess);
+
+  options.onSettled &&= notifyManager.batchCalls(options.onSettled);
+}
+
+export function baseResults<
+  TQueryFnData = unknown,
+  TError = unknown,
+  TData = TQueryFnData,
+  TQueryData = TQueryFnData,
+  TQueryKey extends QueryKey = QueryKey
+>(
+  queryObserver: QueryObserver<
+    TQueryFnData,
+    TError,
+    TData,
+    TQueryData,
+    TQueryKey
+  >,
+  queryKey: TQueryKey,
+  options: DefaultedQueryObserverOptions<
+    TQueryFnData,
+    TError,
+    TData,
+    TQueryData,
+    TQueryKey
+  >,
+  onDestroy: () => void
+): Observable<QueryObserverResult<TData, TError>> {
+  return new Observable((observer) => {
+    const initialResult: QueryObserverResult<TData, TError> = {
+      ...queryObserver.getOptimisticResult({
+        queryKey,
+        ...options,
+      }),
+    };
+
+    observer.next(initialResult);
+
+    const queryObserverDispose = queryObserver.subscribe(
```

Comment:
The only different thing is the observer. You can create one method and pass only the observer. Check how they did it here:
https://github.com/TanStack/query/blob/d018dfa46083ce059e0aadc06b48f729faade825/packages/react-query/src/useInfiniteQuery.ts#L96
https://github.com/TanStack/query/blob/d018dfa46083ce059e0aadc06b48f729faade825/packages/react-query/src/useQuery.ts#L139

Reply:
thank you for example, It did refactor, but could not make a type safe version of `baseQuery` function. They have also casted the types.